src/components/TranslationDialog/TranslationModal/LocalesSelect.js

```diff
@@ -0,0 +1,50 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import { SingleSelect, SingleSelectOption } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+const query = {
+    locales: {
+        resource: 'locales/db',
+    },
+}
+
+export const LocalesSelect = ({ onChange, selected }) => {
+    const { data, fetching } = useDataQuery(query)
+
+    return (
+        <SingleSelect
+            prefix={
+                selected ? i18n.t('Translating to') : i18n.t('Choose a locale')
+            }
+            onChange={({ selected }) => onChange(selected)}
+            loading={fetching}
```

Comment:
```suggestion
            loading={fetching}
            loadingText={i18n.t('Loading locales')}
```
I've looked into showing the selected locale name while the available locales request is pending, but I don't think this is possible, or at least not easy at all. I think your current implementation is probably the best we can do if we want the select to load it's own options.

Alternatively, we could load the locales in the modal, and wait for both the locales and the translations to load before showing the form. Then we can just make this a regular `SingleSelect`.

Both solutions are fine, but for the current implementation I think a loading text would be a good addition.

Reply:
Sorry, I just came to realise the `loadingText` shows up in the dropdown, not in the "input" part of the component. This suggestion doesn't really add much value at all. Please ignore it.